gtfsrt_vehicleposition_denormalized/VehiclePositionService.cs

```diff
@@ -0,0 +1,124 @@
+                                         feed_entity_id = entity.id,
+                                         trip_id = entity.vehicle?.trip?.trip_id,
+                                         route_id = entity.vehicle?.trip?.route_id,
+                                         direction_id = entity.vehicle?.trip?.direction_id,
+                                         trip_start_date = entity.vehicle?.trip?.start_date,
+                                         trip_start_time = entity.vehicle?.trip?.start_time,
+                                         trip_schedule_relationship = entity.vehicle?.trip?.schedule_relationship.ToString(),
+                                         vehicle_id = entity.vehicle?.vehicle?.id,
+                                         vehicle_label = entity.vehicle?.vehicle?.label,
+                                         vehicle_license_plate = entity.vehicle?.vehicle?.license_plate,
+                                         vehicle_timestamp = entity.vehicle?.timestamp,
+                                         current_stop_sequence = entity.vehicle?.current_stop_sequence,
+                                         current_status = entity.vehicle?.current_status.ToString(),
+                                         stop_id = entity.vehicle?.stop_id,
+                                         congestion_level = entity.vehicle?.congestion_level.ToString(),
+                                         occupancy_status = null, //where to get this from?
+                                         latitude = entity.vehicle?.position?.latitude,
+                                         longitude = entity.vehicle?.position?.longitude,
+                                         bearing = entity.vehicle?.position?.bearing,
+                                         odometer = entity.vehicle?.position?.odometer,
+                                         speed = entity.vehicle?.position?.speed
+                                     });
+            }
+
+            InsertVehiclePositionsRows(vehiclePositions);
+        }
+
+        readonly VehiclePositionsDataSet _dataSet = new VehiclePositionsDataSet();
+
+        private void InsertVehiclePositionsRows(List<VehiclePositionData> vehiclePositions)
+        {
+            if (!vehiclePositions.Any())
+            {
+                Log.Debug("No vehicle positions to save...");
+                return;
+            }
+
+            try
+            {
+                Log.Debug($"Trying to insert {vehiclePositions.Count} vehicle position rows in database.");
+                _dataSet.SaveData(vehiclePositions);
+                Log.Debug($"Inserted {vehiclePositions.Count} vehicle position rows in database.");
+            }
+            catch (Exception exception)
+            {
+                Log.Debug($"Failed to save data: {exception.Message}");
+            }
+        }
+        public void Stop()
+        {
+            Thread.Sleep(1000);
+            Environment.Exit(0);
```

Comment:
Does this need to stop `feedMessageThread`?

Reply:
@cbirjega says that this isn't an issue. May not be necessary but doesn't hurt.